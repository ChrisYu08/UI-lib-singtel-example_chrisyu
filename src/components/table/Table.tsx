import * as React from 'react';
import ConciseTable from './ConciseTable';
import type { TableProps } from './ConciseTable';

function Table<RecordType extends object = any>(
  props: TableProps<RecordType>,
) {
  const renderTimesRef = React.useRef(0);
  renderTimesRef.current += 1;

  return <ConciseTable<RecordType> {...props} />;
}

export default Table;
