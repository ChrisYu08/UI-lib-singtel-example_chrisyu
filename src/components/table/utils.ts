
import type { ColumnTitle, ColumnTitleProps, Key } from './tableInterface';

export function renderColumnTitle<RecordType>(
  title: ColumnTitle<RecordType>,
  props: ColumnTitleProps<RecordType>,
) {
  if (typeof title === 'function') {
    return title(props);
  }

  return title;
}