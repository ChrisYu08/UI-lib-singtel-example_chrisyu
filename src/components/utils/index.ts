


export function getPrefixCls(type:string){
  const prefixMap={
    table:'singtel-table'
  }
  return prefixMap[type]
}
