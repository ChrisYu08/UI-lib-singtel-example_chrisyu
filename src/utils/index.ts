export const getBaseWidth = () => {
    const screenWidth = document.body.clientWidth;
    return screenWidth;
};

/*with less than 767*/
export const isMobile = () => {
    const baseWidth = getBaseWidth();
    return baseWidth <= 767;
};