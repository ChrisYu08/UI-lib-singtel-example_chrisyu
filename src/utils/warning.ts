import rcWarning, { resetWarned } from 'rc-util/lib/warning';

export { resetWarned };
export function noop() {}

type Warning = (valid: boolean, component: string, message?: string) => void;

// eslint-disable-next-line import/no-mutable-exports
let warning: Warning = noop;
if (process.env.NODE_ENV !== 'production') {
  warning = (valid, component, message) => {
    rcWarning(valid, `[singtelComponent: ${component}] ${message}`);

    if (process.env.NODE_ENV === 'test') {
      resetWarned();
    }
  };
}

export default warning;