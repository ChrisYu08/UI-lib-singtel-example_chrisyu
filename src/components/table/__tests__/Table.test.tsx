import React from 'react';
import { screen, cleanup } from "@testing-library/react";
import Table from '..';
import {render} from '../../../tests/utils';



describe('Table',()=>{
  afterEach(() => {
    cleanup();
  });
  const data = [
    { name: 'Jack1', age: 38, address: 'some where', key: '1',operations:'qwe' },
    { name: 'Rose', age: 26, address: 'some where', key: '2',operations:'aaa' },
  ];
  it('renders JSX correctly for plain table', () => {
    const columns = [
      {
        title: 'Name',
        dataIndex: 'name',
        key: 'name',
        width: '40%',
      },
      {
        title: 'Age',
        dataIndex: 'age',
        key: 'age',
        width: '20%',
      },
      {
        title: 'Address',
        dataIndex: 'address',
        key: 'address',
        width: '20%',
      },
      {
        title: 'Operations',
        dataIndex: 'operations',
        key: 'operations',
      },
    ];
      const { asFragment } = render(
        <Table data={data} columns={columns} />,
      );
      expect(asFragment().firstChild).toMatchSnapshot();
    }
  )
  it('renders JSX correctly for sorted table', () => {
    const columns = [
      {
        title: 'Name',
        dataIndex: 'name',
        key: 'name',
        width: '40%',
        sorter: (a: { name: string | any[]; }, b: { name: string | any[]; }) => a?.name?.length - b?.name?.length,
        sortDirections: ['descend'],
      },
      {
        title: 'Age',
        dataIndex: 'age',
        key: 'age',
        width: '20%',
        defaultSortOrder: 'descend',
        sorter: (a: { age: number; }, b: { age: number; }) => a.age - b.age,
      },
      {
        title: 'Address',
        dataIndex: 'address',
        key: 'address',
        width: '20%',
      },
      {
        title: 'Operations',
        dataIndex: 'operations',
        key: 'operations',
      },
    ];
    const { asFragment } = render(
      <Table data={data} columns={columns} />,
    );
    expect(asFragment().firstChild).toMatchSnapshot();
  }
)
it('renders JSX correctly for selection table', () => {
  const columns = [
    {
      title: 'Name',
      dataIndex: 'name',
      key: 'name',
      width: '40%',
      sorter: (a: { name: string | any[]; }, b: { name: string | any[]; }) => a?.name?.length - b?.name?.length,
      sortDirections: ['descend'],
    },
    {
      title: 'Age',
      dataIndex: 'age',
      key: 'age',
      width: '20%',
      defaultSortOrder: 'descend',
      sorter: (a: { age: number; }, b: { age: number; }) => a.age - b.age,
    },
    {
      title: 'Address',
      dataIndex: 'address',
      key: 'address',
      width: '20%',
    },
    {
      title: 'Operations',
      dataIndex: 'operations',
      key: 'operations',
    },
  ];
  const rowSelection = {
    onChange: (selectedRowKeys: React.Key[], selectedRows: any[]) => {
      console.log(`selectedRowKeys: ${selectedRowKeys}`, 'selectedRows: ', selectedRows);
    }
  };

  const { asFragment } = render(
    <Table
      rowSelection={{
        type: 'checkbox',
        ...rowSelection,
      }}
      columns={columns} 
      data={data} 
    />,
  );

  expect(asFragment().firstChild).toMatchSnapshot();
  }
)
})