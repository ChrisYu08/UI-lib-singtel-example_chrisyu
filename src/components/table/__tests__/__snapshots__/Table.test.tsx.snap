// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders JSX correctly for plain table 1`] = `
<div
  class="singtel-table-wrapper"
>
  <div
    class="singtel-table"
  >
    <div
      class="singtel-table-container"
    >
      <div
        class="singtel-table-content"
      >
        <table
          style="table-layout: auto;"
        >
          <colgroup />
          <thead
            class="singtel-table-thead"
          >
            <tr>
              <th
                class="singtel-table-cell"
                scope="col"
              >
                Contract details
              </th>
            </tr>
          </thead>
          <tbody
            class="singtel-table-tbody"
          >
            <tr
              class="singtel-table-row singtel-table-row-level-0"
              data-row-key="1"
            >
              <td
                class="singtel-table-cell"
              >
                <ul>
                  <li>
                    <span
                      class="head1"
                    >
                      name:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      Jack1
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      age:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      38
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      address:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      some where
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      operations:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      qwe
                    </span>
                  </li>
                </ul>
              </td>
            </tr>
            <tr
              class="singtel-table-row singtel-table-row-level-0"
              data-row-key="2"
            >
              <td
                class="singtel-table-cell"
              >
                <ul>
                  <li>
                    <span
                      class="head1"
                    >
                      name:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      Rose
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      age:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      26
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      address:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      some where
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      operations:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      aaa
                    </span>
                  </li>
                </ul>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table renders JSX correctly for selection table 1`] = `
<div
  class="singtel-table-wrapper"
>
  <div
    class="singtel-table"
  >
    <div
      class="singtel-table-container"
    >
      <div
        class="singtel-table-content"
      >
        <table
          style="table-layout: auto;"
        >
          <colgroup>
            <col
              style="width: 32px;"
            />
          </colgroup>
          <thead
            class="singtel-table-thead"
          >
            <tr>
              <td
                class="singtel-table-cell"
              />
              <th
                class="singtel-table-cell"
                scope="col"
              >
                Contract details
              </th>
            </tr>
          </thead>
          <tbody
            class="singtel-table-tbody"
          >
            <tr
              class="singtel-table-row singtel-table-row-level-0"
              data-row-key="1"
            >
              <td
                class="singtel-table-cell"
              >
                <div
                  class="chooseColumn"
                >
                  <span>
                    <div>
                      <span
                        class="singtel-input-checkbox-container"
                      >
                        <input
                          class="singtel-input-checkbox"
                          type="checkbox"
                          value="1"
                        />
                        <label
                          for="singtel-input-checkboxId"
                        />
                        <span />
                      </span>
                    </div>
                  </span>
                </div>
              </td>
              <td
                class="singtel-table-cell"
              >
                <ul>
                  <li>
                    <span
                      class="head1"
                    >
                      name:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      Jack1
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      age:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      38
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      address:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      some where
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      operations:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      qwe
                    </span>
                  </li>
                </ul>
              </td>
            </tr>
            <tr
              class="singtel-table-row singtel-table-row-level-0"
              data-row-key="2"
            >
              <td
                class="singtel-table-cell"
              >
                <div
                  class="chooseColumn"
                >
                  <span>
                    <div>
                      <span
                        class="singtel-input-checkbox-container"
                      >
                        <input
                          class="singtel-input-checkbox"
                          type="checkbox"
                          value="2"
                        />
                        <label
                          for="singtel-input-checkboxId"
                        />
                        <span />
                      </span>
                    </div>
                  </span>
                </div>
              </td>
              <td
                class="singtel-table-cell"
              >
                <ul>
                  <li>
                    <span
                      class="head1"
                    >
                      name:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      Rose
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      age:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      26
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      address:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      some where
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      operations:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      aaa
                    </span>
                  </li>
                </ul>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table renders JSX correctly for sorted table 1`] = `
<div
  class="singtel-table-wrapper"
>
  <div
    class="singtel-table"
  >
    <div
      class="singtel-table-container"
    >
      <div
        class="singtel-table-content"
      >
        <table
          style="table-layout: auto;"
        >
          <colgroup />
          <thead
            class="singtel-table-thead"
          >
            <tr>
              <th
                class="singtel-table-cell"
                scope="col"
              >
                Contract details
              </th>
            </tr>
          </thead>
          <tbody
            class="singtel-table-tbody"
          >
            <tr
              class="singtel-table-row singtel-table-row-level-0"
              data-row-key="1"
            >
              <td
                class="singtel-table-cell"
              >
                <ul>
                  <li>
                    <span
                      class="head1"
                    >
                      name:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      Jack1
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      age:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      38
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      address:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      some where
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      operations:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      qwe
                    </span>
                  </li>
                </ul>
              </td>
            </tr>
            <tr
              class="singtel-table-row singtel-table-row-level-0"
              data-row-key="2"
            >
              <td
                class="singtel-table-cell"
              >
                <ul>
                  <li>
                    <span
                      class="head1"
                    >
                      name:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      Rose
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      age:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      26
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      address:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      some where
                    </span>
                  </li>
                  <li>
                    <span
                      class="head1"
                    >
                      operations:
                    </span>
                       
                    <span
                      class="val1"
                    >
                      aaa
                    </span>
                  </li>
                </ul>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
